persistence/rocksdb/src/main/java/org/infinispan/persistence/rocksdb/RocksDBStore.java

```diff
@@ -0,0 +1,547 @@
+        return PersistenceUtil.count(this, null);
+    }
+
+    @Override
+    public boolean contains(Object key) {
+        try {
+            return load(key) != null;
+        } catch (Exception e) {
+            throw new PersistenceException(e);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public void process(KeyFilter keyFilter, CacheLoaderTask cacheLoaderTask, Executor executor, boolean loadValues, boolean loadMetadata) {
+
+        int batchSize = 100;
+        ExecutorAllCompletionService eacs = new ExecutorAllCompletionService(executor);
+        final TaskContext taskContext = new TaskContextImpl();
+
+        List<Entry> entries = new ArrayList<>(batchSize);
+        try {
+            semaphore.acquire();
+        } catch (InterruptedException e) {
+            throw new PersistenceException("Cannot acquire semaphore: CacheStore is likely stopped.", e);
+        }
+        try {
+            if (stopped) {
+                throw new PersistenceException("RocksDB is stopped");
+            }
+            Optional<RocksIterator> optionalIterator = getDBIterator(this.db);
+            try {
+                if (optionalIterator.isPresent()) {
+                    RocksIterator it = optionalIterator.get();
+                    for (it.seekToFirst(); it.isValid(); it.next()) {
+                        Entry entry = new Entry(it.key(), it.value());
+                        entries.add(entry);
+                        if (entries.size() == batchSize) {
+                            final List<Entry> batch = entries;
+                            entries = new ArrayList<>(batchSize);
+                            submitProcessTask(cacheLoaderTask, keyFilter, eacs, taskContext, batch, loadValues, loadMetadata);
+                        }
+                    }
+                    if (!entries.isEmpty()) {
+                        submitProcessTask(cacheLoaderTask, keyFilter, eacs, taskContext, entries, loadValues, loadMetadata);
+                    }
+
+                    eacs.waitUntilAllCompleted();
+                    if (eacs.isExceptionThrown()) {
+                        throw new PersistenceException("Execution exception!", eacs.getFirstException());
+                    }
+                }
+            } catch (Exception e) {
+                throw new PersistenceException(e);
+            } finally {
+                if (optionalIterator.isPresent()) {
+                    optionalIterator.get().close();
+                }
+            }
+        } finally {
+            semaphore.release();
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private void submitProcessTask(final CacheLoaderTask cacheLoaderTask, final KeyFilter filter, CompletionService ecs,
+                                   final TaskContext taskContext, final List<Entry> batch,
+                                   final boolean loadValues, final boolean loadMetadata) {
+        ecs.submit(new Callable<Void>() {
```

Comment:
No, because the callee throws checked exceptions and that is unsupported by lambdas